waltz-ng/client/data-types/resolvers.js

```diff
@@ -0,0 +1,26 @@
+export function loadDataTypes(dataTypeService) {
+    return dataTypeService.loadDataTypes();
+}
+
+loadDataTypes.$inject = ['DataTypeService'];
+
+
+export function dataTypeIdResolver($q, $stateParams, dataTypeService) {
```

Comment:
Fair comment. Will do that.  I think I was still thinking in terms of state
params and simple types

On 12 Aug 2016 18:01, "David Watkins" <notifications@github.com> wrote:

> In waltz-ng/client/data-types/resolvers.js
> <https://github.com/khartec/waltz/pull/423#discussion_r74624692>:
>
> > @@ -0,0 +1,26 @@
> > +export function loadDataTypes(dataTypeService) {
> > +    return dataTypeService.loadDataTypes();
> > +}
> > +
> > +loadDataTypes.$inject = ['DataTypeService'];
> > +
> > +
> > +export function dataTypeIdResolver($q, $stateParams, dataTypeService) {
>
> Maybe I've got this all wrong... But why not have the resolver return the
> data type entity not the id. Then have two resolvers
> , one for byCode and one for byId, both should be simple proxies to the
> service and would remove need for the $q service ?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/khartec/waltz/pull/423/files/d15abf63a870586f0fc2d8c3ad62524e834ebfe8#r74624692>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AC4528q0Adu5t-8_HXe8sNFJmIsPamCZks5qfKbfgaJpZM4JjPwe>
> .
>